contributed_definitions/nyaml/NXtransfer_matrix_table.yaml

```diff
@@ -58,12 +58,13 @@ NXtransfer_matrix_table(NXobject):
     \@output:
       doc: |
         Specific name of output beam which the transfermatrix table is related to.
-    dimensions:
-      doc: |
-        Square matrix with dimension N_variables x N_variables
-      rank: 2
-      dim:
-        [
-          [1, N_variables],        
-          [2, N_variables], 
-        ]
\ No newline at end of file
+    data(NX_NUMBER):

```

Comment:
If TRANSFER_MATRIX is a field (NX_NUMBER) and not a group (NXdata) anymore, it cannot have subfields attached, so the property 'dimensions' shall belong to TRANSFER_MATRIX itself.